File: victim.cpp
#include <iostream>
#include <string>

int main() {
    std::string input;
    if (!std::getline(std::cin, input)) {
        std::cerr << "[victim] stdin closed\n";
        return 1;
    }

    // Trigger 1: deliberate null-dereference -> SIGSEGV
    if (input == "CRASH") {
        std::cerr << "[victim] CRASH -> triggering SIGSEGV\n";
        volatile int *p = nullptr;
        *p = 0xDEAD;
        return 0;
    }

    // Trigger 2: infinite spin -> timeout / SIGKILL
    if (input == "LOOP") {
        std::cerr << "[victim] LOOP -> spinning forever\n";
        volatile bool spin = true;
        while (spin) {}
        return 0;
    }

    std::cout << "Safe: " << input << "\n";
    return 0;
}

File: victim.py
import sys

def main():
    try:
        data = input()
    except EOFError:
        print("[victim] stdin closed", file=sys.stderr)
        sys.exit(1)

    if data == "CRASH":
        print("[victim] CRASH -> triggering ZeroDivisionError", file=sys.stderr)
        # Deliberate unhandled exception — exits with code 1 and a traceback
        result = 1 / 0  # noqa: F841

    elif data == "LOOP":
        print("[victim] LOOP -> spinning forever", file=sys.stderr)
        while True:
            pass  # infinite spin — killed by sandbox timeout

    else:
        print(f"Safe: {data}")

if __name__ == "__main__":
    main()
